### PR TITLE
feat: general maintenance of ci/cd

### PR DESCRIPTION
**Pull request by peterdeme**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            // Source branch looks like this: release-1.0.0
+            const version = context.payload.pull_request.head.ref.split('-')[1]
+            core.exportVariable('VERSION', version)
+
+      - uses: ./.github/actions/setup-node
+
+      - name: Publish package
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
[the remark this answers is not in the document]
Comment (arnautov-anton) [reply .github/workflows/release.yml:35]: @petyosi

**Comment by MartinCupela** [comment]
@peterdeme have you considered using semantic-release package for release automation?

**Review by ferhatelmas — APPROVED** [review]
LGTM

**Comment by ferhatelmas** [comment]
> @peterdeme have you considered using semantic-release package for release automation?

we're using the same pattern for all backend SDKs, changing out of scope of this iteration but could be considered for all in another iteration cc @peterdeme

**Comment by MartinCupela** [comment]
@peterdeme there are 2 failed jobs in the CI. Will you still work on that?

File: .github/workflows/release.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (petyosi) [reply .github/workflows/release.yml:35]: Aded token.

[the remark this answers is not in the document]
Comment (ferhatelmas) [reply .github/workflows/release.yml:35]: @petyosi token is added for https://www.npmjs.com/package/getstream right ? I see auth error

**Pull request merged**